src/Sextant/Navigation/ViewStackServiceBase.cs

```diff
@@ -0,0 +1,237 @@
+
+        /// <summary>
+        /// Pops the <see cref="INavigable" /> off the stack.
+        /// </summary>
+        /// <param name="animate">if set to <c>true</c> [animate].</param>
+        /// <returns>An observable that signals when the pop is complete.</returns>
+        public IObservable<Unit> PopModal(bool animate = true) => View.PopModal().Do(_ => PopStackAndTick(ModalSubject));
+
+        /// <inheritdoc />
+        public IObservable<Unit> PopPage(bool animate = true) => View.PopPage(animate);
+
+        /// <inheritdoc />
+        public IObservable<Unit> PopToRootPage(bool animate = true) => View.PopToRootPage(animate).Do(_ => PopRootAndTick(PageSubject));
+
+        /// <inheritdoc />
+        public IObservable<Unit> PushModal(IViewModel modal, string contract = null)
+        {
+            if (modal == null)
+            {
+                throw new ArgumentNullException(nameof(modal));
+            }
+
+            return View
+                .PushModal(modal, contract)
+                .Do(_ =>
+                {
+                    AddToStackAndTick(ModalSubject, modal, false);
+                    Logger.Debug("Added modal '{modal.Id}' (contract '{contract}') to stack.");
+                });
+        }
+
+        /// <inheritdoc />
+        public IObservable<Unit> PushPage(
+            INavigable viewModel,
+            string contract = null,
+            bool resetStack = false,
+            bool animate = true) => PushPage((IViewModel)viewModel, contract, resetStack, animate);
+
+        /// <inheritdoc />
+        public IObservable<Unit> PushPage(IViewModel viewModel, string contract = null, bool resetStack = false, bool animate = true)
+        {
+            if (viewModel == null)
+            {
+                throw new ArgumentNullException(nameof(viewModel));
+            }
+
+            return View
+                .PushPage(viewModel, contract, resetStack, animate)
+                .Do(_ =>
+                {
+                    AddToStackAndTick(PageSubject, viewModel, resetStack);
+                    Logger.Debug($"Added page '{viewModel.Id}' (contract '{contract}') to stack.");
+                });
+        }
+
+        /// <summary>
+        /// Returns the top modal from the current modal stack.
+        /// </summary>
+        /// <returns>An observable that signals the top modal view model.</returns>
+        [SuppressMessage("Design", "CA1826: Do not use Enumerable methods on indexable collections.", Justification = "Deliberate usage")]
+        public IObservable<IViewModel> TopModal() => ModalSubject.FirstAsync().Select(x => x.Last());
+
+        /// <summary>
+        /// Returns the top page from the current navigation stack.
+        /// </summary>
+        /// <returns>An observable that signals the top page view model.</returns>
+        [SuppressMessage("Design", "CA1826: Do not use Enumerable methods on indexable collections.", Justification = "Deliberate usage")]
```

Comment:
Why not use x[x.Count - 1] instead of suppressing the message.

Reply:
Changed to use count based operation